common/CommonFunctions.ml

```diff
@@ -166,14 +156,56 @@ and get_published_info_class cls =
   get_first_release_string (get_first_release cls.obj_release.internal)
 
 and get_published_info_field field cls =
-  let expRelease = get_prototyped_release_string field.lifecycle in 
+  let expRelease = get_prototyped_release_string field.lifecycle in
   let clsRelease = get_first_release cls.obj_release.internal in 
   let fieldRelease =  get_first_release field.release.internal in
-  let clsReleaseIndex = list_index_of clsRelease DT.release_order in
-  let fieldReleaseIndex = list_index_of fieldRelease DT.release_order in
+  let clsReleaseIndex = list_index_of clsRelease code_name_order in
+  let fieldReleaseIndex = list_index_of fieldRelease code_name_order in
     if (not (expRelease = "")) then
       expRelease
     else if (not (clsReleaseIndex = -1)) && (not (fieldReleaseIndex = -1)) && (clsReleaseIndex < fieldReleaseIndex) then
       get_first_release_string fieldRelease
     else
       ""
+
+and render_template template_file json output_file =
+  let lines_read = Stdext.Unixext.read_lines template_file in
+  let templ = Mustache.of_string (String.concat "\n" lines_read) in
+  let rendered = Mustache.render templ json in
+  let out_chan = open_out output_file in
+  finally (fun () -> fprintf out_chan "%s" rendered)
```

Comment:
Strangely the docs https://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html don't mention an exception thrown from output_string - is it so? Meaning is the finally needed?